core/object/class_db.h

```diff
@@ -185,33 +185,61 @@ class ClassDB {
 	}
 
 	template <class T>
-	static void register_class(bool p_virtual = false) {
+	static void register_class(bool p_virtual = false, bool p_exposed = true) {
 		GLOBAL_LOCK_FUNCTION;
 		static_assert(TypesAreSame<typename T::self_type, T>::value, "Class not declared properly, please use GDCLASS.");
 		T::initialize_class();
 		ClassInfo *t = classes.getptr(T::get_class_static());
 		ERR_FAIL_COND(!t);
 		t->creation_func = &creator<T>;
-		t->exposed = true;
+		t->exposed = p_exposed;
 		t->is_virtual = p_virtual;
 		t->class_ptr = T::get_class_ptr_static();
 		t->api = current_api;
 		T::register_custom_data_to_otdb();
 	}
 
 	template <class T>
-	static void register_abstract_class() {
+	static void register_abstract_class(bool p_exposed = true) {
 		GLOBAL_LOCK_FUNCTION;
 		static_assert(TypesAreSame<typename T::self_type, T>::value, "Class not declared properly, please use GDCLASS.");
 		T::initialize_class();
 		ClassInfo *t = classes.getptr(T::get_class_static());
 		ERR_FAIL_COND(!t);
-		t->exposed = true;
+		t->exposed = p_exposed;
 		t->class_ptr = T::get_class_ptr_static();
 		t->api = current_api;
 		//nothing
 	}
 
+	template <class T>
+	static void register_virtual_class(bool p_exposed = true) {
+		GLOBAL_LOCK_FUNCTION;
+		T::initialize_class();
```

Comment:
```suggestion
		static_assert(TypesAreSame<typename T::self_type, T>::value, "Class not declared properly, please use GDCLASS.");
		T::initialize_class();
```
Same below, see #81020